data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/coordinator/enhanced/EnhancedSourceCoordinator.java

```diff
@@ -116,6 +116,8 @@ public interface EnhancedSourceCoordinator {
      */
     Optional<EnhancedSourcePartition> getPartition(final String partitionKey);
 
+    Optional<EnhancedSourcePartition> getPartition(final String partitionKey, final String partitionType);
```

Comment:
Who is using this and why did you add it?

Reply:
This is needed to give flexibility to get partitions that have types other than GLOBAL

Reply:
If source plugin creators want to get a partition, they have to give it type GLOBAL currently